src/WebJobs.Extensions/Extensions/Timers/Scheduling/CronSchedule.cs

```diff
@@ -32,13 +61,149 @@ internal CrontabSchedule InnerSchedule
         public override bool AdjustForDST => true;
 
-        /// <inheritdoc/>
+        /// <inheritdoc/>        
         public override DateTime GetNextOccurrence(DateTime now)
         {
-            return _cronSchedule.GetNextOccurrence(now);
+            // Note: TimeZoneInfo.Local is mocked in tests
+            return GetNextOccurrence(new DateTimeOffset(now, TimeZoneInfo.Local.GetUtcOffset(now))).LocalDateTime;
+        }
+
+        private DateTimeOffset GetNextOccurrence(DateTimeOffset now)
+        {
+            // The cron library only supports DateTime and does not take TimeZones into account. We need to 
+            // do some manipulating of the time it returns to ensure we're taking the correct action during
+            // transitions into and out of Daylight Saving Time.
+            var nowDateTime = now.LocalDateTime;
+            var nextDateTime = _cronSchedule.GetNextOccurrence(nowDateTime);
+
+            // This will apply the correct offset for the local time zone based on the DateTime.
+            var next = new DateTimeOffset(nextDateTime);
+
+            // Now we need to evaluate the possibility of Daylight Saving transitions.
+            // Example of DST transitions in 2018 for reference (from .NET source):
+            //
+            //         -=-=-=-=-=- Pacific Standard Time -=-=-=-=-=-=-
+            //   March 11, 2018                            November 4, 2018
+            // 2AM            3AM                        1AM              2AM
+            // |      +1 hr     |                        |       -1 hr      |
+            // | <invalid time> |                        | <ambiguous time> |
+            //                  [========== DST ========>)
+            if (TryAdjustInvalidTime(next, out DateTimeOffset? adjusted))
+            {
+                // "Spring Forward": "next" was invalid and was adjusted forward to a valid time.                
+                next = adjusted.Value;
+            }
+            else if (TryAdjustAmbiguousTime(now, next, out adjusted))
+            {
+                // "Fall Back": Either "now" or "next" was ambiguous and the time was adjusted.
+                next = adjusted.Value;
+            }
+
+            return next;
+        }
+
+        /// <summary>
+        /// If the provided DateTime is not valid, moves forward to the next occurrence in the schedule
+        /// until a valid DateTime is found. For example, 02:30 does not exist when "springing forward"
+        /// to Daylight Saving Time in the Pacific Standard Time Zone. Uses TimeZoneInfo.Local to make
+        /// the determination.
+        /// </summary>
+        /// <param name="next">The DateTimeOffset to evaluate.</param>
+        /// <param name="adjusted">The adjusted DateTimeOffset, if "next" is invalid.</param>
+        /// <returns></returns>
+        internal bool TryAdjustInvalidTime(DateTimeOffset next, out DateTimeOffset? adjusted)
```

Comment:
I realized that this was completely unnecessary. Just calling `next.ToLocalTime()` takes care of any invalid time that may have been created. So I've removed the entire method in the next iteration.